### PR TITLE
hsmd: Add validate_commitment_tx

### PR DESCRIPTION
**Pull request by ksedgwic**
Add hsmd_validate_commitment_tx message so validating signers can independently validate the peer's signatures.

[Summary of all changes for validating signers](https://gitlab.com/lightning-signer/docs/-/wikis/C-Lightning-Modifications)

### CODE REVIEW

File: channeld/channeld.c [hunk 1]
```diff
@@ -1564,6 +1587,8 @@ static void handle_peer_commit_sig(struct peer *peer, const u8 *msg)
 	const struct htlc **htlc_map, **changed_htlcs;
 	const u8 *funding_wscript;
 	size_t i;
+	struct simple_htlc **htlcs;
+	const u8 * msg2;
```
Comment (devrandom) [inline channeld/channeld.c:1591]: maybe just `msg`?

**Review by devrandom — APPROVED** [review]
some nits, but LGTM

File: hsmd/hsmd_wire.csv [hunk 2]
```diff
@@ -138,6 +138,21 @@ msgdata,hsmd_sign_commitment_tx,commit_num,u64,
 msgtype,hsmd_sign_commitment_tx_reply,105
 msgdata,hsmd_sign_commitment_tx_reply,sig,bitcoin_signature,
 
+# Validate the counterparty's commitment signatures.
+msgtype,hsmd_validate_commitment_tx,35
```
[the remark this answers is not in the document]
Comment (devrandom) [reply hsmd/hsmd_wire.csv:142]: OK with existing name then

File: channeld/channeld.c (hunk 1 above)
Comment (devrandom) [reply channeld/channeld.c:1591]: I notice that in `handle_peer_revoke_and_ack`, the existing code uses a descriptive name `revocation_msg` to disambiguate.  Probably best to follow that convention.

**Comment by ksedgwic** [comment]
This was merged upstream in https://github.com/ElementsProject/lightning/pull/5108

**Pull request closed without merging**